facilitator/src/gcp_oauth.rs

```diff
@@ -114,17 +335,11 @@ pub(crate) struct GcpOauthTokenProvider {
     logger: Logger,
 }
 
-impl fmt::Debug for GcpOauthTokenProvider {
+impl<P: ProvideAwsCredentials> fmt::Debug for GcpOauthTokenProvider<P> {
```

Comment:
The compiler won't have it:
```
error[E0277]: the trait bound `T: ProvideAwsCredentials` is not satisfied
   --> src/gcp_oauth.rs:338:24
    |
313 | pub(crate) struct GcpOauthTokenProvider<P: ProvideAwsCredentials> {
    |                                            --------------------- required by this bound in `GcpOauthTokenProvider`
...
338 | impl<T> fmt::Debug for GcpOauthTokenProvider<T> {
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ProvideAwsCredentials` is not implemented for `T`
    |
help: consider restricting type parameter `T`
    |
338 | impl<T: ProvideAwsCredentials> fmt::Debug for GcpOauthTokenProvider<T> {
    |       ^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
However this is moot, because as of the latest commit, `GcpOauthTokenProvider` is no longer generic over P.

Reply:
Ohh I suspect this is is because ProvideAwsCredentials is not bounded by Debug. And the `#[derive(Debug)]` logic probably depends on basically "T can be Debug if all fields of T are Debug."